dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java

```diff
@@ -871,6 +875,24 @@ public Grid substituteMetaData( int sourceColumnIndex, int targetColumnIndex, Ma
         return this;
     }
 
+    @Override
+    public Grid substituteValues( Map<String, String> uidToNameMap )
+    {
+        for ( List<Object> columns : grid )
+        {
+            for ( int col = 0; col < columns.size(); col++ )
+            {
+                Object column = columns.get( col );
+                if ( column != null && !MathUtils.isNumeric( String.valueOf( column ) ) )
+                {
+                    columns.set( col, uidToNameMap.getOrDefault( String.valueOf( column ), String.valueOf( column ) ) );

```

Comment:
Regarding 

`uidToNameMap.getOrDefault( String.valueOf( column ), String.valueOf( column ) )`

is this intentional? is the default the same as the initial value?

Reply:
Fixed